.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,88 @@
+name: Build and Publish Marketplace
+on:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    container: cimg/go:1.19
+    defaults:
+      run:
+        shell: bash
+    steps:
+    - name: build/checkout-repo
+      uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+      with:
+        fetch-depth: 0
+    - name: build/ensure-master-exists
+      run: git rev-parse --verify master >/dev/null 2>&1 || git branch master origin/master
+    - name: build/check-style
+      run: make check-style
+    - name: build/test
+      run: make test
+    - name: build/build
+      if: ${{ github.repository == 'mattermost/mattermost-marketplace' && (github.ref_name != 'private-beta') }}
+      run: make build
+    - name: build/build-private-beta
```

Comment:
Added build target for `private-beta` that essentially just sets `BUILD_UPSTREAM_URL`.